src/main/java/net/mcreator/element/types/VillagerProfession.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2022, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.element.types;
+
+import net.mcreator.element.GeneratableElement;
+import net.mcreator.element.parts.MItemBlock;
+import net.mcreator.element.parts.Sound;
+import net.mcreator.io.FileIO;
+import net.mcreator.ui.workspace.resources.TextureType;
+import net.mcreator.util.FilenameUtilsPatched;
+import net.mcreator.workspace.elements.ModElement;
+
+import java.io.File;
+
+public class VillagerProfession extends GeneratableElement {
+
+	public String name;
+	public String displayName;
+	public MItemBlock oldPOI;
```

Comment:
Why do you need POI twice and why so different naming of these two?

Reply:
You're right maybe I can rename it, but the oldPoi is to store the saved POI. If you take a deeper look, you'll see it's to check that when the user saves the mod element, the point of interest (profession block) is not already used. If it is, then it checks that this is the same.

Reply:
Done! Waiting for more review...

Reply:
Not sure how this prevents one from making another ME with same POI by storing old value in the file.

Also you could just store old value in a variable in GUI editor, no need to overload definition of GE with saving two values at the same time just for the sake of using it in the ME UI

Reply:
Done!